osx32/build

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+
+CompilerOptions="-DDFRE_OSX32 "
+CompilerOptions+="-nostdlib "
+CompilerOptions+="-std=c++11 "
```

Comment:
You could add this to the linux build file too if you want. What was the error you got without it anyway?

Reply:
I was running into errors with brace initialization in the code without passing `-std=c++11`, specifically errors such as:

```
In file included from ../code/main.cpp:11:
In file included from ../code/posix_platform.cpp:69:
../code/mem_arena.h:72:14: error: expected expression
    *Arena = {};
             ^
In file included from ../code/main.cpp:16:
../code/parser.cpp:125:27: error: expected ';' at end of declaration
    chunk_bounds LastChunk{NFA_NULLSTATE, NFA_STARTSTATE};
                          ^
                          ;
../code/parser.cpp:129:22: error: expected ';' at end of declaration
    lexer_state Lexer{Regex};
                     ^
                     ;

...
```

Using: 

```
% clang --version                                                                                                    
Apple LLVM version 9.0.0 (clang-900.0.39.2)
Target: x86_64-apple-darwin16.7.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

Reply:
Apparently mine is C++17 by default.
```
$ g++ --version
g++ (GCC) 8.3.0
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

$ g++ -dM -E -x c++  /dev/null | grep -F __cplusplus

#define __cplusplus 201402L
```

Try out that second command. I got it from https://stackoverflow.com/a/44735016

Reply:
Oh and I think it would be good to add it to the linux build file. Other people might not be using the newest version of g++

Reply:
.

Reply:
Cool 👍 think we should just address build setting stuff in a separate commit or PR. Looks like MSVC isn't setting a standard too? I'll probably update stuff to be consistent and make sure git tracks that the build scripts should be directly executable too.